libtenzir/include/tenzir/format/syslog.hpp

```diff
@@ -250,6 +254,134 @@ struct message_parser : parser_base<message_parser> {
+      const auto* f = sv.begin();
+      const auto* const l = sv.end();
+      return p(f, l, unused) && f == l;
+    };
+    const auto is_time = [&](const std::string& time) -> bool {
+      const auto hour_parser
+        = integral_parser<uint16_t, 2, 2>{}.with([](uint16_t hour) {
+            return hour <= 23;
+          });
+      const auto minsec_parser
+        = integral_parser<uint16_t, 2, 2>{}.with([](uint16_t x) {
+            return x <= 59;
+          });
+      const auto p
+        = hour_parser >> ':' >> minsec_parser >> ':' >> minsec_parser;
+      auto sv = std::string_view{time};
+      const auto* f = sv.begin();
+      const auto* const l = sv.end();
+      return p(f, l, unused) && f == l;
+    };
+    const auto timestamp_parser
+      = (word.with(is_month) >> ws >> word.with(is_day) >> ws
+         >> word.with(is_time))
+          ->*[](std::tuple<std::string, std::string, std::string, std::string,
+                           std::string>
+                  x) {
+                return fmt::to_string(fmt::join(x, ""));
+              };
+    const auto tag_parser = +(parsers::printable - ':');
+    const auto content_parser = ~(wsignore >> *parsers::printable);
+    // clang-format off
+    const auto p = ~(priority_parser >> wsignore)
+                   >> timestamp_parser >> wsignore    // timestamp
+                   >> word >> wsignore                // host
+                   >> tag_parser >> ':'               // tag
+                   >> content_parser;                 // content
+    // clang-format on
+    if constexpr (std::is_same_v<Attribute, unused_type>)
+      return p(f, l, unused);
+    else
+      return p(f, l, x.timestamp, x.host, x.tag, x.content);
+  }
+};
+
+inline type make_syslog_type() {
+  return type{
+    "syslog.rfc5424",
+    record_type{{
+      {"facility", uint64_type{}},
+      {"severity", uint64_type{}},
+      {"version", uint64_type{}},
+      {"ts", type{"timestamp", time_type{}}},
+      {"hostname", string_type{}},
+      {"app_name", string_type{}},
+      {"process_id", string_type{}},
+      {"message_id", string_type{}},
+      // TODO: The index is currently incapable of handling map_type. Hence, the
+      // structured_data is disabled.
+      // {"structered_data", map_type{
+      //   string_type{}.name("id"),
+      //   map_type{string_type{}.name("key"),
+      //   string_type{}.name("value")}.name("params")},
+      // },
+      {"message", string_type{}},
+    }},
+  };
+}
+
+inline type make_legacy_syslog_type() {
+  return type{"syslog.rfc3164", record_type{{{"facility", uint64_type{}},
+                                             {"severity", uint64_type{}},
+                                             {"timestamp", string_type{}},
+                                             {"host", string_type{}},
+                                             {"tag", string_type{}},
+                                             {"content", string_type{}}}}};
```

Comment:
Is there a reason to not choose the same name for the `timestamp`and `host` fields?

Reply:
Nothing major, but since these fields are not in the same format as in the newer RFC, I picked different names for them. New-RFC `ts` is a `time_type`, old-RFC `timestamp` is a `string_type` containing a wonky timestamp. `hostname` is strictly defined by the new-RFC, but old-RFC `host` seems to also sometimes be wonky (like the `hostname/a.b.c.d` we had in our example data). I can also change them to have the same name, since the schema name is already a differentiating factor.

Reply:
You bring up some valid points. I'm not sure if changing types is better with or without changing field names. Perhaps we have to see how this will play out in practice. I'd assume most people will only receive a single syslog type (though it's great to have the functionality for accepting both!). Having said that, I think renaming to `hostname` is fine, given that the RFC calls this field `HOSTNAME` (but I agree that it is wonky). I would also rename the new-RFC `ts` to `timestamp`. But that's just my opinion, go with whatever you prefer :+1:

Reply:
Yeah, I'll do that.